tests/cdrouter_bootstrap.py

```diff
@@ -58,6 +58,16 @@ def runTest(self):
 
         c = CDRouter(self.cdrouter_server)
 
+        """Go to cdrouter frequency and reboot to Dual-stack config"""
+        wan_ip = board.get_interface_ipaddr("wan0")
+        if self.config.board['cdrouter_location'] == 'CBN':
+            cm_freq = self.config.board['cdrouter_freq']
+        else:
+            cm_freq = self.config.board['common']['CM_online_freq']
+        wan.sendline("snmpset -v 2c -c public -t 30 -r 3 "+wan_ip+" "+board.mib["lastKnownFreq"]+".0 i "+cm_freq)
+        assert 0 != wan.expect(['Timeout: No Response from'] + prompt)
+        board.reprovision(provisioner, cm_cfg="9_EU_CBN_Dual-Stack_LG.txt", mta_cfg="AutoCA.txt")
```

Comment:
This is all board specific... it need to be generic. If you need to do something board specific pre and post run you can add a hook... or possibly just having a test that inherits and runs the provisioning code before starting CDrouter...
```
class myCDrouterTest(CDrouterBootStrap):
   def runTest(self):
       board.provision()
       super(runTest())
```

Reply:
Understood, running the provisioning code before testing cdrouter.